src/policy/libstandardness.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <policy/libstandardness.h>
+
+#include <chainparams.h>
+#include <common/args.h>
+#include <common/shared_lib_utils.h>
+#include <node/blockstorage.h>
+#include <node/kernel_notifications.h>
+#include <policy/fees.h>
+#include <policy/fees_args.h>
+#include <timedata.h>
+#include <validation.h>
+
+namespace {
+
+inline int set_result(libstandardness_result* ret, libstandardness_result sret)
+{
+    if (ret)
+        *ret = sret;
+    return 0;
+}
+
+} // namespace
+
+int libstandard_verify_transaction(const unsigned char *txTo, unsigned int txToLen, libstandardness_result* result)
+{
+    try {
+        TxInputStream stream(PROTOCOL_VERSION, txTo, txToLen);
+        CTransaction tx(deserialize, stream);
+
+        const auto args = ArgsManager{};
+        //TODO: pass the chain type as a library argument
+        const auto chainParams = CreateChainParams(args, ChainType::MAIN);
+
+        // We initialize block manager options for the dummy chainstate.
+        node::BlockManager::Options blockman_opts {
+            .chainparams = *chainParams,
+            .blocks_dir = args.GetBlocksDirPath(),
+        };
+
+        // We initialize mempool options and mempool for the dummy chainstate.
+        const auto fee_estimator = std::make_unique<CBlockPolicyEstimator>(FeeestPath(args));
+        //TODO: give an adequate value for internal mempool consistency check
+        CTxMemPool::Options mempool_opts {
+            .estimator = fee_estimator.get(),
+            .check_ratio = 0,
+        };
+
+        const auto dummy_mempool = std::make_unique<CTxMemPool>(mempool_opts);
+
+        node::KernelNotifications notifications{};
+        ChainstateManager::Options chainman_opts{
+            .chainparams = *chainParams,
+            .datadir = args.GetDataDirNet(), 
+            .adjusted_time_callback = GetAdjustedTime,
+            .notifications = notifications,
+        };
+        const auto chainman = std::make_unique<ChainstateManager>(chainman_opts, blockman_opts);
+
+        const auto dummy_chainstate = std::make_unique<Chainstate>(dummy_mempool.get(), chainman->m_blockman, *chainman);
+        //TODO: initialize dummy_chainstate with chain tip as seen by the caller and spent_UTXO
```

Comment:
As of `adc0921ea1`, I think this is very unclear what validation API is offered by libbitcoinkernel. There is no `doc/libbitcoinerkernel` so far and `bitcoin-chainstate.cpp` is just for demo-only purpose so far.

Verifying the propagation of transactions for use-cases like Lightning and other contracting protocols require more API flexibility than “validation-at-the-tip” as feerates might have to be disregard (i.e they can be added latter with a CPFP) and absolute / relative timelocks ignored (user A funds safety is defined by the ability to broadcast a _valid_ transaction after X though strictly not before X to avoid jeopardizing user B funds safety e.g the htlc-claim / htlc-timeout flow).

Additionally, resources-constrained clients like LN mobile might still validate the propagation validity of their transactions while relying on SPV servers (bip157, electrum) for the chainstate, and therefore uncoupled validation libraries give them more flexibility.